### DISCUSSION TITLE
Creating new textures

### DISCUSSION THREAD

**Discussion by Moburma** [swfans/syndwarsfx#223] [General]
Not sure if this should be an issue rather than a discussion. I guess more for the attention of the honourable gentleman @mefistotelis  

I've tried to generate new raw texture graphics with the rips I made from the PS1 version, but also things like the unused textures in the Japanese .fli videos. Broadly speaking this does work, but there are weird problems with certain colours where I'm not sure if it's an incompatibility with Syndicate Wars versus Dungeon Keeper, or I'm just doing it wrong.

An example is this. I use this input png file
![](https://i.postimg.cc/MHT5KQd9/PS1-TEX01f.png)

And run with this argument:
.\pngpal2raw.exe -f raw -p .\PAL.PAL -o .\TEX01.DAT .\TEX01ps1.png

This works, and it makes me a new texture file. This does indeed work in-game, but you get weird colour glitches, mostly around the lights in buildings. Some example pictures:

![](https://i.postimg.cc/0ypYvn98/Desktop-Screenshot-2024-11-24-21-48-26-32.png)

![](https://i.postimg.cc/hGc9tmGV/Desktop-Screenshot-2024-11-24-21-49-33-72.png)

I tried messing about with the various other options back in the day with it, but never got anywhere. Anything obvious I'm doing wrong?

PAL.PAL is just the one from the DATA folder.

I've tried it with other source images I've made and had the same problems with them, usually different specific colours.

**Comment by mefistotelis** [comment]
You're treating the image collection as one image - that is the issue. note that in KeeperFX the images are separated, stored in a way where tiles match each other. Then the tool itself glues them together.

For a quick fix, it should work better if you disable diffusion. But to get best quality of conversion, you have to provide the tool with separate tiles to glue.

**Reply by Moburma** [reply]
Thanks, I didn't realise that. I ran it with -l 0 and that mostly fixed it, but still some issues with the transparent balconies. I'll try and do it the proper way later. Is there any documentation on the format of the input text files?

![](https://i.postimg.cc/R04nttxT/Desktop-Screenshot-2024-11-25-09-39-29-98.png)

I also discovered a strange thing I didn't fully appreciate until now. In the Japanese version of the game, TEX04.DAT is exactly the same as the original game's file (this contains mostly vehicle related textures). However, the DEMOx.FLI files uses a version of this for what I thought was just padding. The Japanese one has a different set of new TEX04 textures in there.

Only I realise now this is NOT just padding and is not unused - in fact the game doesn't use TEX04.DAT for the vehicle textures, it uses a frame from DEMO-10.fli instead! So it **is** used by the Japanese version, and replacing that file in the other versions results in the same new textures in-game too.

**Reply by mefistotelis** [reply]
>  Is there any documentation on the format of the input text files?

No, you need to read the code to figure that out.

> n fact the game doesn't use TEX04.DAT for the vehicle textures, it uses a frame from DEMO-10.fli instead!

I think that depends on settings, at least on PC.
